modules/basis/src/lib/parse-basis.js

```diff
@@ -1,8 +1,8 @@
 import {loadBasisModule} from './basis-module-loader';
 
 async function parse(data, options) {
-  const module = loadBasisModule();
-  return module;
+  const {BasisFile} = await loadBasisModule(options);
+  return new BasisFile(new Uint8Array(data));
```

Comment:
You should move the extraction code in here, under different options.

Reply:
I will.